src/Parsers/ZetaSQL/ASTZetaSQLQuery.h

```diff
@@ -0,0 +1,32 @@
+#pragma once
+
+
+#include "Parsers/IAST.h"
+#include "Parsers/IAST_fwd.h"
+namespace DB::ZetaSQL
+{
+
+    class ASTZetaSQLQuery: public IAST
+    {
+    public:
+        enum class StageKeyword
+        {
+           FROM,
+           SELECT,
+           AGGREGATE,
+           WHERE,
+        };
+
+        struct PipelineStage
+        {
+            StageKeyword type;
+            ASTPtr expression;
+        };
+
+        std::list<PipelineStage> stages;
```

Comment:
`std::list<PipelineOperators> operators`? That would be the official syntax at least.